docs/nais-application/migration.md

```diff
@@ -0,0 +1,192 @@
+          echo ${GITHUB_TOKEN} | docker login ghcr.io -u ${{github.actor}} --password-stdin
+          docker build -t ${IMAGE} .
+          docker push ${IMAGE}
+
+  deploy:
+    name: Deploy to NAIS
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: nais/deploy/actions/deploy@v1
+        env:
+          APIKEY: ${{ secrets.NAIS_DEPLOY_APIKEY }}
+          CLUSTER: target-cluster
+          RESOURCE: my-resource/.nais/nais-cluster.yaml
+          VAR: image=${{ env.IMAGE }}
+```
+
+Initially, eliminate the environment variable: `IMAGE: ghcr.io/navikt/my-team/my-app:${{ github.sha }}`, as
+the `nais/docker-build-action`, automatically generates this value and outputs it as the `image` variable.
+
+```yaml
+europe-north1-docker.pkg.dev/nais-management-233d/my-team/my-app
+```
+
+The resulting image reference is composed of the following:
+
+* Registry: `europe-north1-docker.pkg.dev`
+* Google Cloud Project: `nais-management-233d`
+* Team: `my-team`
+* Application: `my-app`
+
+The tag is automatically generated by the `nais/docker-build-action` or can be specified as input to
+the [action](https://github.com/nais/docker-build-push).
+
+If you wish to refer to the image later, please consult the guide
+on [Image registry](https://doc.nais.io/guides/application/#step-6-push-your-image-to-our-image-registry).
+
+* Substitute the preceding code block and replace it with the following:
+
+```yaml
+  - name: Build and publish Docker image
+    if: ithub.ref_name == 'main'
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    run: |
+      echo ${GITHUB_TOKEN} | docker login docker.pkg.github.com -u ${GITHUB_REPOSITORY} --password-stdin
+      echo ${GITHUB_TOKEN} | docker login ghcr.io -u ${{github.actor}} --password-stdin
+      docker build -t ${IMAGE} .
+      docker push ${IMAGE}
+```
+
+```yaml
+  - name: Push docker image to GAR and sign image
+    uses: nais/docker-build-push@v0
+    id: docker-build-push
+    with:
+      team: my-team
+      identity_provider: ${{ secrets.NAIS_WORKLOAD_IDENTITY_PROVIDER }} # Provided as Organization Secret
+      project_id: ${{ vars.NAIS_MANAGEMENT_PROJECT_ID }} # Provided as Organization Variable
+```
+
+* In the "deploy" step, substitute the given image reference, with the output of the previous step. The output of the
+  previous step is the `image` variable, given the example over it can be referenced as
+  follows; `${{ steps.docker-build-push.outputs.image }}`. The id `docker-build-push` is the id of the previous step,
+  there is where your new image will be outputted. In this example we devided our workflow into two jobs, `build`
+  and `deploy`. The `deploy` job depends on the `build` job, and will not run unless the `build` job is successful.
+  The `build` job outputs the image reference as `image`, and the `deploy` job can reference this output
```

Comment:
This part hasn't been shown yet, it is only shown in the final workflow at the bottom. That can be confusing.